src/Symfony/Component/Translation/Tests/TranslatorTest.php

```diff
@@ -602,6 +600,56 @@ public function testMissingLoaderForResourceError()
 
         $translator->getCatalogue('en');
     }
+
+    public function testTransWithGlobalParameters()
+    {
+        $translator = new Translator('en');
+        $translator->addLoader('array', new ArrayLoader());
+        $translator->addResource('array', ['welcome' => 'Welcome {name}!'], 'en');
+        $translator->addResource('array', ['welcome' => 'Bienvenue {name}!'], 'fr');
+        $translator->addGlobalParameter('{name}', 'Global name');
+
+        $this->assertSame('Welcome Global name!', $translator->trans('welcome'));
+        $this->assertSame('Bienvenue Global name!', $translator->trans('welcome', [], null, 'fr'));
+        $this->assertSame('Welcome John!', $translator->trans('welcome', ['{name}' => 'John']));
+        $this->assertSame('Bienvenue Jean!', $translator->trans('welcome', ['{name}' => 'Jean'], null, 'fr'));
+    }
+
+    public function testTransWithGlobalTranslatableParameters()
+    {
+        $translator = new Translator('en');
+        $translator->addLoader('array', new ArrayLoader());
+        $translator->addResource('array', ['welcome' => 'Welcome on {link}!'], 'en');
+        $translator->addResource('array', ['welcome' => 'Bienvenue sur {link}!'], 'fr');
+
+        $translator->addResource('array', ['url' => 'example.com/admin'], 'en', 'globals');
+        $translator->addResource('array', ['url' => 'example.fr/admin'], 'fr', 'globals');
+
+        $translator->addGlobalTranslatableParameter('{link}', 'url', [], 'globals');
+
+        $this->assertSame('Welcome on example.com/admin!', $translator->trans('welcome'));
+        $this->assertSame('Bienvenue sur example.fr/admin!', $translator->trans('welcome', [], null, 'fr'));
+        $this->assertSame('Welcome on other.com!', $translator->trans('welcome', ['{link}' => 'other.com']));
+        $this->assertSame('Bienvenue sur autre.fr!', $translator->trans('welcome', ['{link}' => 'autre.fr'], null, 'fr'));
+    }
+
+    /**
+     * @requires extension intl
```

Comment:
Does this really requires the extension ? We support the ICU format without the extension thanks to our polyfill package.

Reply:
If we can use the polyfill, we may remove this requirement from all tests, right?

https://github.com/symfony/symfony/blob/6f4f04b98f6134996a57d21d1851d558a34dc45c/src/Symfony/Component/Translation/Tests/TranslatorTest.php#L561